src/features/site-survey/actions.ts

```diff
@@ -214,6 +220,49 @@ export const updateModifiedFeatures = (
     }
   });
 
+/**
+ * Action that adjusts home positions to the current drone positions
+ * if possible.
+ */
+export const adjustHomePositionsToDronePositions =
+  (): AppThunk => async (dispatch, getState) => {
+    // Only outdoor shows are supported by the dialog.
+
+    const distanceFunction = haversineDistance;
+    const homePositions = getHomePositionsInWorldCoordinates(getState());
+    const dronePositions = getCurrentGPSPositionsOfActiveUAVs(getState());
+    if (homePositions === undefined || dronePositions.includes(undefined)) {
+      return;
+    }
+
+    const distances = calculateDistanceMatrix(
+      homePositions,
+      dronePositions as GPSPosition[], // Validated in guard clause above.
+      {
+        distanceFunction,
+        // eslint-disable-next-line @typescript-eslint/ban-types
+        getter: (item: GPSPosition | null): LonLat =>
```

Comment:
Can we ever receive `null` here? We excluded `undefined` above - maybe we should do the same with `null`?

Reply:
I think we technically can, but I'll recheck. Since there are several calls and results here that come from untyped JS code, in one or two places I just preferred to be "safe" with the typing.

Reply:
I added an `isGPSPositionValid()` type guard that does `null`, `undefined`, and null island checks only, so we don't do unnecessary calculation with `isGPSPosition()`. I also added the null island check to `isGPSPosition()` for consistency. It'd be nice to get a confirmation on whether this is okay.